crates/nargo_cli/src/preprocess.rs

```diff
@@ -0,0 +1,57 @@
+use acvm::ProofSystemCompiler;
+use iter_extended::vecmap;
+use noirc_driver::{CompiledContract, CompiledProgram};
+
+// TODO: migrate to `nargo_cli`
```

Comment:
Can this be done before merging?

Reply:
I'm guessing this should read `nargo-core`. Yeah, I was planning on addressing this in a followup PR which moves a bunch of similar logic over in one go but I can address this now (or reverse the order of the PRs).